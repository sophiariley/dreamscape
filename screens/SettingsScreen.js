import { StyleSheet, View, Text, Pressable, Alert } from "react-native";
import {Feather} from 'react-native-vector-icons';

const showAlert = (navigation) =>
  Alert.alert(
    'Log Out',
    'Are you sure you want to log out?',
    [
      {text: 'Yes', onPress: () => navigation.navigate('Login')},
      {text: 'No', style: 'cancel'}
    ]
  );


const SettingsScreen = ({navigation}) => {
    return (
        <View style={styles.container}>
            <Pressable style={styles.logoutBox}
                onPress={() => showAlert(navigation)}>
                <Text style={styles.text}>Logout</Text>
                <Feather name='arrow-right' style={styles.arrow} size={20}/>
            </Pressable>
        </View>
    )
}

export default SettingsScreen;

const styles = StyleSheet.create({
    container: {
        backgroundColor: 'white',
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'column'
    },
    logoutBox: {
        width: '100%',
        height: '15%',
        flexDirection: 'row',
        alignContent: 'center',
        justifyContent: 'space-around',
        backgroundColor: 'rgba(217,217,217, 0.05)',
        borderWidth: 1,
        borderColor: 'rgba(217,217,217, 0.7)',
    },
    text: {
        fontSize: 20,
        alignSelf: 'center',
        color: '#3A6496',
        opacity: .8,
        right: 50
    },
    arrow: {
        alignSelf: 'center',
        color: '#3A6496',
        opacity: .8,
        left: 50
    },
})