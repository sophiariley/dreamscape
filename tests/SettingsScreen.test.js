import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native'; 
import SettingsScreen from './SettingsScreen';
import {create} from 'react-test-renderer';

// basic rendering test
test('renders correctly across screens', () => {
  const tree = create(<SettingsScreen />);
  expect(tree).toMatchSnapshot();
});


// // Testing login button
// describe('Testing login button', () => {
//     it('should find the button via testId', () => {
//     const testIdName = 'loginButton';
//     const {getByTestId} = render(<LoginScreen />);
//     const foundButton = getByTestId(testIdName);
//     expect(foundButton).toBeTruthy();
//     });

//     it('button should say login', () => {
//         const loginText = ' Login! '; // spaces around Login! because that's how it appears in code
//         const notFoundText = 'error :(';
//         const {toJSON, getByText, queryByText} = render(<LoginScreen />);
//         const foundLoginTextElement = getByText(loginText);
//         const notFoundTextElement = queryByText(notFoundText);
//         expect(foundLoginTextElement.props.children).toEqual(loginText);
//         expect(notFoundTextElement).toBeNull();
//         expect(toJSON()).toMatchSnapshot();
//       });

//       it('login button is pressable', () => {
//         const page = render(<LoginScreen/>);
//         const loginButton = page.getByTestId('loginButton');
//         fireEvent.press(loginButton);
//       })
// });

// // Testing user input boxes

// const USERNAME_TEXT = 'username';

// describe('Testing username input', () => {
//   it('should find the input box via testId', () => {
//   const testIdName = 'usernameInput';
//   const {getByTestId} = render(<LoginScreen />);
//   const foundButton = getByTestId(testIdName);
//   expect(foundButton).toBeTruthy();
//   });

//     it('Input box is pressable', () => {
//       // const navigation = {navigate: jest.fn()};
//       // jest.spyOn(navigation, 'navigate');
//       const page = render(<LoginScreen/>);
//       const usernameInput = page.getByTestId('usernameInput');
//       fireEvent.press(usernameInput);
//       // expect(navigation.navigate).toHaveBeenCalledTo("Home");
//     });

//     test('Input box should allow for input', () => {
//       render(<LoginScreen/>);
//       fireEvent.changeText(screen.getByPlaceholderText('Username'), USERNAME_TEXT)
//     })
// });

// // Testing user input boxes

// const PASSWORD_TXT = 'password';

// describe('Testing password input', () => {
//   it('should find the password box via testId', () => {
//   const testIdName = 'passwordInput';
//   const {getByTestId} = render(<LoginScreen />);
//   const foundButton = getByTestId(testIdName);
//   expect(foundButton).toBeTruthy();
//   });

//     it('Password box is pressable', () => {
//       // const navigation = {navigate: jest.fn()};
//       // jest.spyOn(navigation, 'navigate');
//       const page = render(<LoginScreen/>);
//       const usernameInput = page.getByTestId('passwordInput');
//       fireEvent.press(usernameInput);
//       // expect(navigation.navigate).toHaveBeenCalledTo("Home");
//     });

//     test('Password box should allow for input', () => {
//       render(<LoginScreen/>);
//       fireEvent.changeText(screen.getByPlaceholderText('Password'), PASSWORD_TXT)
//     })
// });


// // Testing forgot password pressable

// describe('Testing forgot password', () => {
//   it('should find forgot password via testId', () => {
//   const testIdName = 'forgotPassword';
//   const {getByTestId} = render(<LoginScreen />);
//   const foundButton = getByTestId(testIdName);
//   expect(foundButton).toBeTruthy();
//   });

//   it('Forgot password is pressable', () => {
//     const navigation = {navigate: () =>{}}
//     const page = render(<LoginScreen navigation={navigation}/>);
//     const forgotPassword = page.getByTestId('forgotPassword');
//     fireEvent.press(forgotPassword);
//   });
  
//   it('Forgot password navigates to reset password screen', () => {
//     const navigation = {navigate: () =>{}}
//     const page = render(<LoginScreen navigation={navigation}/>);
//     jest.spyOn(navigation, 'navigate')
//     const forgotPassword = page.getByTestId('forgotPassword');
//     fireEvent.press(forgotPassword);
//     expect(navigation.navigate).toHaveBeenCalledWith("Reset Password")
//   });
// })


// // Testing create account pressable

// describe('Testing create account', () => {
//   it('should find create account via testId', () => {
//   const testIdName = 'createAccount';
//   const {getByTestId} = render(<LoginScreen />);
//   const foundButton = getByTestId(testIdName);
//   expect(foundButton).toBeTruthy();
//   });

//   it('Create account is pressable', () => {
//     const navigation = {navigate: () =>{}}
//     const page = render(<LoginScreen navigation={navigation}/>);
//     const createAccount = page.getByTestId('createAccount');
//     fireEvent.press(createAccount);
//   });
  
//   it('Create account navigates to create account screen', () => {
//     const navigation = {navigate: () =>{}}
//     const page = render(<LoginScreen navigation={navigation}/>);
//     jest.spyOn(navigation, 'navigate')
//     const createAccount = page.getByTestId('createAccount');
//     fireEvent.press(createAccount);
//     expect(navigation.navigate).toHaveBeenCalledWith("Create Account 1")
//   });
// })
