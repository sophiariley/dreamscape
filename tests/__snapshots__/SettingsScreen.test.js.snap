// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignContent": "center",
        "backgroundColor": "rgba(217,217,217, 0.05)",
        "borderColor": "rgba(217,217,217, 0.7)",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": "15%",
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
    testID="logoutButton"
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "#3A6496",
          "fontSize": 20,
          "opacity": 0.8,
          "right": 50,
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;
